scripts/concurrent_task_utils.py

```diff
@@ -42,13 +44,55 @@ def log(message, show_time=False):
             python_utils.PRINT(message)
 
 
+class TaskResult(python_utils.OBJECT):
+    """Task result for concurrent_task_utils."""
+
+    def __init__(
+            self, name, failed, trimmed_messages, messages,
+            report_enabled=True):
+        """Constructs a TaskResult object.
+
+        Args:
+            name: str. The name of the task.
+            failed: bool. The boolean value representing whether the task
+                failed.
+            trimmed_messages: list(str). List of error messages that are
+                trimmed to keep main part of messages.
+            messages: list(str). List of full messages returned by the objects.
+            report_enabled: bool. Decide whether task result will print or not.
+        """
+        self.name = name
+        self.failed = failed
+        self.trimmed_messages = trimmed_messages
+        self.messages = messages
+        self.report_enabled = report_enabled
```

Comment:
This should be a property of the `TaskThread` object instead of `TaskResult`.

Reply:
@DubeySandeep If I do this then it would not be possible to add the status message in get_report function because this boolean help to decide if it has to add pass/fail status or not

Reply:
If the `report_enabled` is true for taskThread object then log using `get_report` else log the `self.messages`, is that helpful?

Reply:
Done.